reascripts/ReaSpeech/source/ReaSpeechWorker.lua

```diff
@@ -280,15 +291,19 @@ function ReaSpeechWorker:check_active_job_request_status()
   end
 end
 
-function ReaSpeechWorker:check_active_job_transcript_request_status()
+function ReaSpeechWorker:handle_responses()
   local active_job = self.active_job
-  local request = active_job.transcript_request
-
-  if request:ready() then
-    self:handle_response(active_job, request:result())
-    self.active_job = nil
-  elseif request:error() then
-    self:handle_error(active_job, request:error())
-    self.active_job = nil
+  active_job.responses = {}
+  for _, request in pairs(active_job.requests) do
+    if request:ready() then
+      table.insert(active_job.responses, request:result())
+    elseif request:error() then
+      self:handle_error(active_job, request:error())
+      self.active_job = nil
+      return
+    end
```

Comment:
extract responses from the request objects, stopping if any of them are `error`s.

Reply:
> Not sure if there are potential use cases where we'd want to continue processing in spite of some errors. I've seen batched request/response models (1:1 between requests and responses) where each response can be an error, and it's up to the caller to decide if the existence of an error is fatal, or if should handle partial failure.

yeah, that had occurred to me as well. the "multiple result objects" parts of this pr were/are largely on spec i think after you'd mentioned the idea of being able to do stem separation. i didn't go as deep there because it felt a bit out of reach even for a demo plugin at this point.

a background goal for this pr was for all of the changes to be under-the-hood but to set us up to hit the ground running with other task ideas.